packages/dev/docs/pages/blog/building-a-combobox.mdx

```diff
@@ -0,0 +1,143 @@
+setting up a node filter to determine if a node should be left visible, hidden, or skipped in the case where its parent was already hidden. In addition, we watch for any changes in the DOM while the menu is open
+via a [MutationObserver](https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver), hiding those new elements if need be. When the menu closes, every node that we modified is reverted back to its previous state.
+Check out our `ariaHideOutside` hook in `@react-aria/overlays` if you'd like to learn more.
+
+### Mobile implementation divergence
+
+Another divergence between the WAI-ARIA 1.2 ComboBox spec and our ComboBox implementation came as a result of our mobile implementation. Initially, we considered the outer input and the tray input as both being individual combo boxes that mirror each others state. This became increasingly complicated as we tried to
+manage things like focus and internal state between the two inputs, resulting in several tray input specific affordances added to our `useComboBox` hook, reducing the aria hook's generality by introducing this React Spectrum specific behavior. Interacting with the tray input with a screen reader was also problematic
+since the screen reader would detect that the input had `role="combobox"` and thus erroneously state "double tap to close", a touch action usually reserved for focusing an input. After some deliberation, we decided to alter the accessibility implementation for the mobile experience, removing the outer combo box input in favor for a button that
+looks like a combo box. Tapping the button opens the tray that contains the actual combo box input, albeit with `role="searchbox"` instead of `role="combobox"` to avoid the aforementioned screen reader announcement.
+
+### Differences in screen readers
+
+Just like browsers, screen readers can be quite varied in behavior and may require deliberate tooling to ensure a consistent degree of functionality. While automated accessibility tools can be useful for ensuring that your component meets baseline accessibility guidelines,
+manual testing is irreplaceable for surfacing these behavioral differences, illustrated in part by the image below.
+
+```tsx import
+import screenreaderSpreadsheetImageUrl from 'url:../assets/screenreader-spreadsheet-combobox.png';
+import {Image, ImageContext, ImageModal} from '@react-spectrum/docs';
+
+ReactDOM.render((
+  <ImageContext.Provider value="">
+    <ImageModal>
+      <Image src={screenreaderSpreadsheetImageUrl} alt="Screenshot of ComboBox screen reader vs browser test matrix" style={{maxWidth: 'min(100%, 780px)', display: 'block', margin: '20px auto', cursor: 'zoom-in'}} />
+      <Image src={screenreaderSpreadsheetImageUrl} alt="Screenshot of ComboBox screen reader vs browser test matrix" style={{maxWidth: 'initial'}} />
+    </ImageModal>
+  </ImageContext.Provider>
+), document.getElementById('spreadsheet-image'));
```

Comment:
shoot looks like this doesn't work in the built version, gonna try doing something with portals perhaps